### PR TITLE
LinalgRefactor-SGVector-Rangefill

### PR DESCRIPTION
**Pull request by OXPHOS**
Only works for cpu backend.

### CODE REVIEW

File: tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc [hunk 1]
```diff
@@ -17,3 +17,43 @@ TEST(LinalgBackendEigen, dot)
...
+{
+	SGVector<float64_t> v(4);
+	range_fill(v, 0.0);
+	EXPECT_EQ(v.vlen, 4);
+	for(int32_t i=0;i<4;i++)
+		EXPECT_NEAR(v[i], i, 1e-9);
+}
+
+//test when start>0
+TEST(LinalgBackendEigen, eigen3_backend_start_greaterThan_0)
```
Comment (karlnapf) [inline tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc]: pls dont mix camelCase with camel_case

File: tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc [hunk 2]
```diff
@@ -17,3 +17,43 @@ TEST(LinalgBackendEigen, dot)
...
+{
+	SGVector<float64_t> v(4);
+	range_fill(v, 5.0);
+	EXPECT_EQ(v.vlen, 4);
+	for(int32_t i=0;i<4;i++)
+		EXPECT_NEAR(v[i], i+5.0, 1e-9);
+}
+
+//test for negative values
+TEST(LinalgBackendEigen, eigen3_backend_start_lessThan_0)
```
Comment (karlnapf) [inline tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc]: no need for "eigen3_backend" in test name as this is already in the right file

Comment (karlnapf) [inline tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc]: Can also change that for the already merged tests and in the other backends

**Comment by karlnapf** [comment]
Fine to merge from my side

If you want to, you can send another patch where you remove range_fill from SGVector, and then replace all the compile errors with sg_linalg

**Comment by OXPHOS** [comment]
lemme see whether i can fix it now. shouldn't be too hard.

**Comment by vigsterkr** [comment]
@OXPHOS what happening?

**Comment by OXPHOS** [comment]
@vigsterkr ready to merge

**Pull request merged**